test/metabase/query_processor_test/nested_queries_test.clj

```diff
@@ -491,77 +479,80 @@
                :limit        10
                :filter       [:> *sender_id/Integer 3]}))))))
 
-(deftest native-query-with-default-params-as-source-test
+(deftest ^:parallel native-query-with-default-params-as-source-test
   (testing "make sure using a native query with default params as a source works"
-    (is (= {:query  "SELECT \"source\".* FROM (SELECT * FROM PRODUCTS WHERE CATEGORY = ? LIMIT 10) AS \"source\" LIMIT 1048575"
-            :params ["Widget"]}
-           (t2.with-temp/with-temp [Card card {:dataset_query {:database (mt/id)
-                                                               :type     :native
-                                                               :native   {:query         "SELECT * FROM PRODUCTS WHERE CATEGORY = {{category}} LIMIT 10"
-                                                                          :template-tags {:category {:name         "category"
-                                                                                                     :display_name "Category"
-                                                                                                     :type         "text"
-                                                                                                     :required     true
-                                                                                                     :default      "Widget"}}}}}]
+    (qp.store/with-metadata-provider (lib.tu/metadata-provider-with-cards-for-queries
+                                      meta/metadata-provider
+                                      [{:database (meta/id)
```

Comment:
not for this large pr, but I would prefer to see mocked cards constructed with the mlv2 api so we can more freely change the shapes of these data structures.

Reply:
definitely. Once we start porting the QP to manipulate pMBQL rather than legacy I think we'll want to update the tests to use MLv2 to construct the queries as well